### ISSUE TITLE
Expose `_NamespacePath`

### ISSUE THREAD

**Issue by FFY00** [python/cpython#119668]
# Feature or enhancement

### Proposal:

Currently, `_NamespacePath` is prefixed as a private type, but since it's exposed, it should be marked as public API. I have a specific case in mind where I want to do an instance check.

I vaguely remember a discussion somewhere with @warsaw where this was mentioned, but I can't find it.

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

### Linked PRs
* gh-119669

**Comment by warsaw** [comment]
I'll have to look around to see if I can dig up the previous discussion, but +1!

**Comment by ericvsmith** [comment]
I don't think we thought it should be hidden, rather there was just no reason to expose it. What's the need for accessing it? If it's at all compelling, then I'm +1.

**Comment by FFY00** [comment]
For my specific case, I am loading data from a namespace package, or rather, I need to point to the directories containing said data. This could also be solved by `importlib.resources` in `MultiplexedPath`, I have opened https://github.com/python/importlib_resources/issues/309 too.
